src/Microsoft.ML.Transforms/Text/NgramUtils.cs

```diff
@@ -200,4 +201,18 @@ private bool ProcessSkipNgrams(int icol, int i, int skips)
             return more;
         }
     }
+
+    internal static class NgramUtils
+    {
+        public static bool IsValidNgramRawType(Type rawType)
+        {
+            // Can only accept key types that can be converted to U4 (uint).
+            return rawType == typeof(sbyte)
+                || rawType == typeof(byte)
+                || rawType == typeof(short)
+                || rawType == typeof(ushort)
+                || rawType == typeof(int)
+                || rawType == typeof(uint);
```

Comment:
>|| rawType == typeof(uint); [](start = 16, length = 27)

since the condition was: type.ItemType.RawKind > DataKind.U4, should it accept string?

Reply:
I inverted the condition. Before, the code was checking for `RawKind > DataKind.U4` and returning `false` Now, this check is checking for "valid" types - ones that are `<= DataKind.U4`. (Although, I did make a change that `0` and negative were "valid" before, but they are no longer.)